packages/flutter_tools/test/commands/run_test.dart

```diff
@@ -28,6 +43,75 @@ void main() {
       }
     });
 
+    testUsingContext('dart-flags option is not available on stable channel', () async {
+      when(mockDeviceManager.getDevices()).thenAnswer((Invocation invocation) {
+        return Stream<Device>.fromIterable(<Device>[
+          FakeDevice(),
+        ]);
+      });
+
+      final RunCommand command = TestRunCommand();
+      final List<String> args = <String> [
+                                  'run',
+                                  '--dart-flags', '"--observe"',
+                                  '--no-hot',
+                                ];
+
+      // Stable branch.
+      try {
+        await createTestCommandRunner(command).run(args);
+        fail('Expect exception');
+      // ignore: unused_catch_clause
+      } on UsageException catch(e) {
+        // Not available while on stable branch.
+      }
+    }, overrides: <Type, Generator>{
+      DeviceManager: () => mockDeviceManager,
+      FlutterVersion: () => mockStableFlutterVersion,
+    });
+
+    testUsingContext('dart-flags option is only populated for debug and profile modes', () async {
+      when(mockDeviceManager.getDevices()).thenAnswer((Invocation invocation) {
+        return Stream<Device>.fromIterable(<Device>[
+          FakeDevice(),
+        ]);
+      });
+      final RunCommand command = TestRunCommand();
+      final List<String> args = <String> [
+                                  'run',
+                                  '--dart-flags', '"--observe"',
+                                  '--no-hot',
+                                ];
+
+      // Debug mode
+      try {
+        await createTestCommandRunner(command).run(args);
+        fail('Expect exception');
```

Comment:
Isn't passing --dart-flags supposed to be allowed in Debug?

Reply:
`FakeDevice.startApp` is setup to throw `ToolExit` with the exit code set to 1 on success, -1 on unexpected result.

Reply:
Maybe add some constants to FakeDevice to make this more self-documenting.

Reply:
Done.